game/scripts/vscripts/Sections/Chapter4/SectionWards.ts

```diff
@@ -0,0 +1,178 @@
+import * as tg from "../../TutorialGraph/index";
+import * as tut from "../../Tutorial/Core";
+import { getOrError, getPlayerHero, printEventTable } from "../../util";
+import { RequiredState } from "../../Tutorial/RequiredState";
+import { TutorialContext } from "../../TutorialGraph/index";
+
+enum NeutralGoalKeys {
+    FetchWards,
+    PlaceObserverWard,
+    PlaceSentryWard
+}
+
+enum GoalState {
+    Started,
+    Completed,
+}
+
+const sectionName: SectionName = SectionName.Chapter4_Wards;
+
+let graph: tg.TutorialStep | undefined = undefined;
+
+const requiredState: RequiredState = {
+    heroLocation: GetGroundPosition(Vector(1900, -3500), undefined)
+};
+const markerLocation = Vector(1000, -4100);
+
+const getGoals = (context: TutorialContext) => {
+    const isGoalStarted = (key: NeutralGoalKeys) =>
+        context[key] === GoalState.Started ||
+        context[key] === GoalState.Completed;
+    const isGoalCompleted = (key: NeutralGoalKeys) =>
+        context[key] === GoalState.Completed;
+
+    const goals: Goal[] = [];
+    const addGoal = (key: NeutralGoalKeys, text: string) => {
+        if (isGoalStarted(key)) {
```

Comment:
I am fine with keeping it like this since this makes sense for other sections, but in this section it seems a little convoluted to track the status of each goal: there can only be one goal active at a time as far as I can see, so this could also just be handled by a single `currentGoal` variable and a `switch` statement.

Reply:
@Perryvw This logic is valid for a lot of other sections. We might want to make an issue and implement it properly for all sections.